mantidproject/mantid
# Add CMake export header feature to more targets (Part II)

**Description of work.**

Further work on this task. See #34482. This PR extends the work to the Geometry, Kernel, MDAlgorithms and Nexus targets

**To test:**

No functional change so just code review and check the Jenkins jobs all pass

Although lots of files changed, the only interesting edits are to the CMakeLists.txt files. The remaining edits are mainly changes to use the MANTID_\<TargetName\>_DLL macro instead of the DLLExport macro

I've also deleted a .cmake file from the Mantid repository that is actually part of the cmake install (GenerateExportHeader.cmake). Not sure why this was duplicated in the Mantid code base but having done a diff vs the cmake version it's the same.

Fixes #27061.

*This does not require release notes* because **no functional change**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
